experimental/dds/tree2/src/simple-tree/proxies.ts

```diff
@@ -815,23 +815,66 @@ function createMapProxy<TSchema extends MapNodeSchema>(
  * @param customTargetObject - Target object of the proxy.
  * If not provided `{}` is used for the target.
  * @remarks
- * Because this proxy is backed by a raw node, it has the same limitations as the node created by {@link createRawObjectNode}.
- * Most if its properties and methods will error if read/called.
+ * Because this proxy is backed by a raw node, it has the same limitations as the node created by {@link createRawNode}.
+ * Most of its properties and methods will error if read/called.
  */
-export function createRawObjectProxy<TSchema extends ObjectNodeSchema>(
+export function createRawNodeProxy<TSchema extends ObjectNodeSchema>(
+	schema: TSchema,
+	content: InsertableTypedNode<TSchema>,
+	allowAdditionalProperties: boolean,
+	target?: object,
+): Unhydrated<TreeObjectNode<TSchema>>;
+export function createRawNodeProxy<TSchema extends FieldNodeSchema>(
+	schema: TSchema,
+	content: InsertableTypedNode<TSchema>,
+	allowAdditionalProperties: boolean,
+	target?: object,
+): Unhydrated<TreeListNodeOld<TSchema["info"]["allowedTypes"]>>;
+export function createRawNodeProxy<TSchema extends MapNodeSchema>(
 	schema: TSchema,
 	content: InsertableTypedNode<TSchema>,
 	allowAdditionalProperties: boolean,
 	target?: object,
-): Unhydrated<TreeObjectNode<TSchema>> {
+): Unhydrated<TreeMapNode<TSchema>>;
+export function createRawNodeProxy<TSchema extends TreeNodeSchema>(
+	schema: TSchema,
+	content: InsertableTypedNode<TSchema>,
+	allowAdditionalProperties: boolean,
+	target?: object,
+): Unhydrated<TreeNode> {
 	// Shallow copy the content and then add the type name symbol to it.
-	const contentCopy = { ...content };
-	Object.defineProperty(contentCopy, typeNameSymbol, { value: schema.name });
-	const proxy = createObjectProxy(schema, allowAdditionalProperties, target);
-	const editNode = createRawObjectNode(schema, contentCopy);
+	let editNode: RawTreeNode<TSchema, InsertableTypedNode<TreeNodeSchema>>;
+	let proxy: TreeNode;
+	if (schema instanceof ObjectNodeSchema) {
+		const contentCopy = copyContent(schema.name, content as InsertableTypedNode<typeof schema>);
+		editNode = createRawNode(schema, contentCopy);
+		proxy = createObjectProxy(schema, allowAdditionalProperties, target);
+	} else if (schema instanceof FieldNodeSchema) {
+		const contentCopy = copyContent(schema.name, content as InsertableTypedNode<typeof schema>);
+		editNode = createRawNode(schema, contentCopy);
+		proxy = createListProxy(allowAdditionalProperties, target);
+	} else if (schema instanceof MapNodeSchema) {
+		const contentCopy = copyContent(schema.name, content as InsertableTypedNode<typeof schema>);
+		editNode = createRawNode(schema, contentCopy);
+		proxy = createMapProxy(allowAdditionalProperties, target);
+	} else {
+		fail("Unrecognized content schema");
+	}
+
 	return setEditNode(proxy, editNode);
 }
 
+function copyContent<T extends object>(typeName: TreeNodeSchemaIdentifier, content: T): T {
+	const copy =
```

Comment:
Nit and possibly just Josh opinion: an if-elseif-else pattern might be nicer here. Would give a place to document assumptions about the input types and make it a bit more obvious where new type support would need to be slotted in in the future. Curious what others think though. Feel free to disregard :)

Reply:
I am fine with if-else, but then I would either have to make `copy` be a `let`, because it is used below all cases, or copy the Object.define property into every case. Not the end of the world, but neither are double-ternaries IMO (so long as prettier does a good job, like this), so I'll just leave it for now. If the type checking gets more complicated in the future then I could see it being obviously better to do if/else.